csharp/src/Ice/Connection.cs

```diff
@@ -331,50 +331,50 @@ public async ValueTask HeartbeatAsync(IProgress<bool>? progress = null, Cancella
             progress?.Report(true);
         }
 
-        /// <summary>Sets a close callback on the connection. The callback is called by the connection when it's
-        /// closed. If the callback needs more information about the closure, it can call Connection.throwException.
-        /// </summary>
-        /// <param name="callback">The close callback object.</param>
-        public void SetCloseCallback(Action<Connection> callback)
+        /// <summary>This event is raised when the connection is closed. If the connection is already closed when
+        /// subscribing the event is raised immediately. If the subscriber needs more information about the closure, it
+        /// can call Connection.ThrowException.</summary>
+        public event EventHandler? Closed
         {
-            lock (_mutex)
+            add
             {
-                if (_state >= ConnectionState.Closed)
+                lock (_mutex)
```

Comment:
You only the locking for the connection state, not for the _closed event.

Reply:
Should we consider throwing ObjectDisposedException instead of calling the event?

Reply:
can we just ignore the subscriber, for example using the default implementation that lets you subscribe, but as the connection is closed you will never receive a new close event?